### DISCUSSION TITLE
How to use the config system?

### DISCUSSION THREAD

**Discussion by simoncozens** [fonttools/fonttools#2834] [General]
I need to turn off Harfbuzz repacking when compiling a troublesome font. I know there's a config option for this, but I don't really understand how to access the fonttools config system.

**Comment by anthrotype** [comment]
you'd set the relevant option in the `TTFont.cfg` dict-like object. E.g. 

https://github.com/fonttools/fonttools/blob/bf6078af3e836e99e741debf3aa8a805ff288b53/Lib/fontTools/subset/__init__.py#L9

and 

https://github.com/fonttools/fonttools/blob/bf6078af3e836e99e741debf3aa8a805ff288b53/Lib/fontTools/subset/__init__.py#L3072

You don't have to actually import the option from the module that use it, you can alternatively use the option's qualified name, in this case it's `"fontTools.ttLib.tables.otBase:USE_HARFBUZZ_REPACKER"`

https://github.com/fonttools/fonttools/blob/bf6078af3e836e99e741debf3aa8a805ff288b53/Lib/fontTools/ttLib/tables/otBase.py#L24

**Reply by anthrotype** [reply]
All the options are globally defined in https://github.com/fonttools/fonttools/blob/main/Lib/fontTools/config/__init__.py

**Reply by anthrotype** [reply]
So to answer your direct question, to force disable using harfbuzz repacker (even when uharfbuzz can be imported), you want to set your font's cfg like this:

```python
font.cfg["fontTools.ttLib.tables.otBase:USE_HARFBUZZ_REPACKER"] = False
```

**Reply by belluzj** [reply]
I think what's missing is a way to pass these options on the command line or using a config file like Behdad was suggesting at some point?

**Reply by anthrotype** [reply]
yeah. I started adding support for that in ufo2ft in https://github.com/googlefonts/ufo2ft/pull/607 a few months ago. Once that in place, fontmake could configure TTFont.cfg through ufo2ft API to set any relevant options, including that one.
